CONTRIBUTING.rst

```diff
@@ -0,0 +1,133 @@
+
+Contributing plugins
+--------------------
+
+If you have written your custom plugin for Brigade there's a good chance that it can be useful for others as well. General guidelines when writing plugins are:
+
+- Make them as generic as possible, it doesn't help others if they only work in your environment
+- Make sure that it's possible to have unit tests which automatically test that the plugins are working
+
+
+Contributing to the Brigade core
+--------------------------------
+
+When you are contributing code to the core of Brigade make sure that the existing tests are passing, and add tests to the code you have added. Having your tests in place ensures that other won't accidentally brake it in the future.
+
+Before you make any significant code changes to the core it's recommended that you open an issue to discuss your ideas before writing the code.
+
+Setting up your environment
+---------------------------
+
+Before contributing for the first time
+
+- Git: If you don't have git installed get it `here <https://git-scm.com/>`_
+- Configure git with your `username and email <https://help.github.com/articles/setting-your-username-in-git/>`_
+- Create a `Github account <https://github.com/join>`_
+- `Fork the repo <https://github.com/brigade-automation/brigade#fork-destination-box>`_
+- Clone your copy:
+
+.. code-block:: bash
+
+	git clone git clone https://github.com/{username}/brigade
+	cd brigade
+
+- Create a new branch:
+
+.. code-block:: bash
+
+	git checkout -b new_stuff
+
+- Create a virtualenv: It's recommended that you do your coding in a `virtualenv <http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_
+- Install Brigade in editable mode
+
+.. code-block:: bash
+
+	pip install -e .
+
+- Commit and push your local changes
+
+.. code-block:: bash
+
+	git add .
+	git commit -m "Meaningful commit message"
+	git push
+
+- Create pull request, visit `the repo <https://github.com/brigade-automation/brigade>`_ and choose the link to create a new pull request.
+
+Running tests
+-------------
+
+While the automated tests will be triggered when you submit a new pull request it can still save you time to run the tests locally first. 
+
+.. code-block:: bash
+
+	pip install tox
+	tox
+
+The test above will run the tests against the Brigade code. Aside from the code the documentation generated by `Sphinx <http://www.sphinx-doc.org/>`_ is also validated. In order to test this you will be required to also install `Pandoc <https://pandoc.org/installing.html>`_. 
+
+Coding style
+------------
+
+Brigade uses `Black <https://github.com/ambv/black>`_, the the uncompromising Python code formatter. Black makes it easy for you to format your code as you can do so automatically after installing it. Note that Python 3.6 is required to run Black.
+
+.. code-block:: bash
+
+	pip install  black
+	black [source_code_directory]
```

Comment:
`black .` should do. Again, we can also hide this in the `make tests` or `make format`.